test/jdk/javax/swing/JButton/bug4323121.java

```diff
@@ -30,96 +30,94 @@
+    static JButton button;
+
+    static volatile Point buttonCenter;
+
+    private static final CountDownLatch windowGainedFocus = new CountDownLatch(1);
+
+    private static final CountDownLatch mouseEntered = new CountDownLatch(1);
+
+    // Usage of this flag is thread-safe because of using the mouseEntered latch
+    private static boolean modelArmed;
 
     public static void main(String[] args) throws Exception {
         Robot robot = new Robot();
         robot.setAutoDelay(100);
+
         try {
             SwingUtilities.invokeAndWait(() -> {
+                button = new JButton("gotcha");
+                button.addMouseListener(new MouseAdapter() {
+                    @Override
+                    public void mouseEntered(MouseEvent e) {
+                        if (button.getModel().isArmed()) {
+                            modelArmed = true;
+                        }
+                        mouseEntered.countDown();
+                    }
+                });
+
                 frame = new JFrame("bug4323121");
-                button = new testButton("gotcha");
                 frame.getContentPane().add(button);
+
+                frame.addWindowFocusListener(new WindowAdapter() {
+                    @Override
+                    public void windowGainedFocus(WindowEvent e) {
+                        windowGainedFocus.countDown();
+                    }
+                });
+
                 frame.pack();
                 frame.setLocationRelativeTo(null);
                 frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
                 frame.setVisible(true);
             });
+
+            if (!windowGainedFocus.await(1, SECONDS)) {
+                throw new RuntimeException("Window didn't gain focus");
+            }
             robot.waitForIdle();
-            robot.delay(1000);
+
             SwingUtilities.invokeAndWait(() -> {
-                pt = button.getLocationOnScreen();
-                buttonW = button.getSize().width;
-                buttonH = button.getSize().height;
+                Point location = button.getLocationOnScreen();
+                buttonCenter = new Point(location.x + button.getWidth() / 2,
+                                         location.y + button.getHeight() / 2);
             });
-            robot.mouseMove(pt.x + buttonW / 2, pt.y + buttonH / 2);
-            robot.waitForIdle();
-            if (failed) {
-                throw new RuntimeException("Any created button returns " +
-                                    "true for isArmed()");
+
+            robot.mouseMove(buttonCenter.x, buttonCenter.y);
+
+            if (!mouseEntered.await(1, SECONDS)) {
+                throw new RuntimeException("Mouse entered event wasn't received");
+            }
+            if (modelArmed) {
+                throw new RuntimeException("getModel().isArmed() returns true "
+                                           + "for a subclass of JButton");
```

Comment:
`summary` and `exception` need to update since you have removed the **subclass of JButton**.